lightbox_css: Update lightbox css to fix UI issues of media previews

getmeapencil opened the pull request:
This PR fixes issues and improves css code of `lightbox` which is responsible for media previews.

---
<details>
<summary>Quick Overview</summary>

| Before | After |
| ------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------- |
| Issue: Misplaced Open button in smaller screens for YouTube preview. Large gap below image list at the bottom. | Fix: UI remain consistent for smaller screens. |
| ![image](https://github.com/user-attachments/assets/91af83a1-f1db-459a-a10b-34ed3576ca9c) | ![image](https://github.com/user-attachments/assets/80f154cd-5f30-4b35-9ee9-2e91e225138c) |
| Issue: The Action buttons are moved down in smaller screen for image preview. | Fix: Buttons stay in their initial position. The name gets truncated. The arrow buttons are vertically more close to center. |
| ![image](https://github.com/user-attachments/assets/999435d8-a80a-47da-89cf-ed97447c1c82) | ![image](https://github.com/user-attachments/assets/adc44edd-90a3-48a5-a969-ed3b0fe27ce8) |
| Issue: The image list goes out of screen for video preview. | Fix: UI remains consistent throughout all types of preview. |
| ![image](https://github.com/user-attachments/assets/b35c32b6-7c6d-4029-9025-7223ae0c6823) | ![image](https://github.com/user-attachments/assets/2e9e815b-6bda-4485-b5e8-6c679c3f6cc0) |
| Layout Before: | Layout After: |
| ![image](https://github.com/user-attachments/assets/678c41eb-da85-40ed-a4b0-43e09bfd7686) | ![image](https://github.com/user-attachments/assets/d31ca722-cbdc-45ca-b908-ba46fa0f3452) |

</details>

---

<details>
<summary>On Smaller Screen (320x747) (After)</summary>
<br>

The media-description gets hidden for Image preview on very small screen considering [available design from CZO](https://chat.zulip.org/#narrow/channel/101-design/topic/UI.20redesign.3A.20image.20viewer/near/1624502).
It is set to not display if the container width (not screen width) is less than 100px. For more than 100px its truncated.

| YT Preview                           | Image Preview                        | Video Preview                        |
| ------------------------------------ | ------------------------------------ | ------------------------------------ |
| ![image](https://github.com/user-attachments/assets/77b8fa8a-29af-4996-9a9f-5bf68fc89bec) | ![image](https://github.com/user-attachments/assets/ad4c1491-2178-44fa-a529-1a18d8428b33) | ![image](https://github.com/user-attachments/assets/7b6d5b3b-238b-42bc-8acf-88187a74ea76) |

</details>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks! Something went wrong with your image uploads, though.
![Screenshot 2025-01-08 at 13 36 09@2x](https://github.com/user-attachments/assets/3f12cd1e-4c01-4789-b54b-10fcd90a2d3c)

alya commented:
Is there a related issue or development community discussion?

getmeapencil commented:
![image](https://github.com/user-attachments/assets/ef3083b1-ae99-46ec-b14f-de8bfd2a1437)

I have re-uploaded the images. Please recheck and confirm.

getmeapencil commented:
> Is there a related issue or development community discussion?

I managed to find this at [CZO](https://chat.zulip.org/#narrow/channel/9-issues/topic/.F0.9F.93.82.20youtube.20preview.20in.20tall.20window/near/1809509).

timabbott commented on web/styles/lightbox.css:
```diff
@@ -32,9 +35,9 @@
+        flex: 1;
         display: flex;
         width: 100%;
-        height: calc(100% - 65px - 95px);
```
I'm definitely glad to be rid of these hardcoded numbers.

timabbott commented on web/styles/lightbox.css:
```diff
@@ -54,6 +54,7 @@
 
         color: hsl(0deg 0% 100% / 80%);
         font-size: 2rem;
+        line-height: 1.8rem;
```
We probably will as a follow-up want to change these from `rem` to `em`, now that we have the "compact mode" font size setting.

karlstolley replied:
I'd go further and say that there should be a more robust use of flexbox for the actions buttons and exit button. Right now, this line-height is used to eyeball centering, but that could easily be improved, especially if the `x` here were replaced with our new closing X icon used around the UI.

karlstolley replied:
(A parallel follow-up would be to introduce icons for the arrows below, too, as right now those are just stretched out carets (`>`).

getmeapencil replied:
> I'd go further and say that there should be a more robust use of flexbox for the actions buttons and exit button. Right now, this line-height is used to eyeball centering, but that could easily be improved, especially if the `x` here were replaced with our new closing X icon used around the UI.

I tried looking around for the instances of close button. I found `×` is being used in places such as channel settings. Since its text, line-height is being used at those instances too. Am I required to make the change?

Also I am not sure if I should be the one changing `rem` to `em` here. Let me know, I'll be happy to make the necessary updates.

karlstolley replied:
Sure. This'll be a follow-up; I've made a note to file an issue for these things.

karlstolley replied:
#33068 for the follow-ups.

karlstolley commented on web/styles/lightbox.css:
```diff
@@ -154,10 +154,10 @@
+            aspect-ratio: 16/9;
+            width: 100%;
+            max-width: $lg_min;
+            max-height: calc(100vh - 200px);
```
I would like to see comments for the rationales on `$lg_min` and the `200px` here.

getmeapencil replied:
Well I don't have any particular reason for using `lg_min` here other than it looked good to me for the screen sizes I checked for responsivity. I have tried explaining the effect of it using comments.

karlstolley commented:
I've posted a few minor concerns, but this looks really good, @getmeapencil. You're welcome to @-message me here when you've addressed them, and then I think we can mark this for integration review.

karlstolley commented:
Ah, regrettably, I've discovered a pretty bad regression on mobile and tablet screens. I believe it traces back to `height: 100vh` being set on `#lightbox_overlay`. Viewport units behave very differently on mobile screens where a browser's tabs and other chrome can be shown or hidden. Right now, the controls at the bottom are shown too low, and cannot be scrolled into view:

| On `main` | This PR |
| --- | --- |
| ![main](https://github.com/user-attachments/assets/b89d4c53-7b5b-4af7-a508-5320eb9db467) | ![this-PR](https://github.com/user-attachments/assets/ec774cba-951e-49a2-bc80-6f098bf49a99) |

Note that things also shift around in the absence of other browser tabs:

| On `main` | This PR |
| --- | --- |
| ![main-no-tabs](https://github.com/user-attachments/assets/39c95542-a79b-4bb0-93da-fa66b65cbf12) | ![this-PR-no-tabs](https://github.com/user-attachments/assets/9da9c788-57c0-4783-98f3-0ba641471f3e) |

karlstolley commented:
Note that it may be possible to resolve this with some of the newer [viewport units](https://developer.mozilla.org/en-US/docs/Web/CSS/length#relative_length_units_based_on_viewport).

getmeapencil commented:
> Note that it may be possible to resolve this with some of the newer [viewport units](https://developer.mozilla.org/en-US/docs/Web/CSS/length#relative_length_units_based_on_viewport).

Thanks for the suggestion. I wasn't able to reproduce this issue but I am hoping using dvh should fix that. Please verify if it has solved the issue.

getmeapencil commented on web/styles/lightbox.css:
```diff
@@ -115,13 +115,8 @@
-        margin-top: 25px;
-        margin-bottom: 15px;
+        flex: 1;
+        min-width: 0;
```
Thanks for pointing it out. I should have added it from start.

getmeapencil commented:
@karlstolley Thanks for the review. 

Changes made-
- used `dvh` instead of `vh`
- added a comment about using `min-width` in `media-description`
- added comments explaining the effect of using `max-width` and `max-height` for iframe

karlstolley commented:
Looks great. Thanks, @getmeapencil -- I'm marking this for integration review.

timabbott commented:
Merged, thanks @getmeapencil and @karlstolley!
